linkerd/stack/src/fallback.rs

```diff
@@ -172,3 +199,57 @@ where
         }
     }
 }
+
+// === impl ReadyFallback ===
+
+impl<N, T, A, B, P, Req> tower::Service<Req> for ReadyFallback<N, T, A, B, P>
+where
+    A: tower::Service<Req>,
+    A::Error: Into<Error>,
+    N: NewService<T, Service = B>,
+    B: tower::Service<Req, Response = A::Response>,
+    B::Error: Into<Error>,
+    P: Fn(&Error) -> bool,
+{
+    type Response = A::Response;
+    type Error = Error;
+    type Future = Either<A::Future, B::Future>;
+
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        // Fallback may only be set until the initial state is polled to ready.
+        match self.fallback {
+            None => self.inner.poll_ready(cx),
+            ref mut fallback => {
+                debug_assert!(
+                    matches!(self.inner, Either::A(_)),
+                    "must not be in fallback yet"
+                );
+
+                match futures::ready!(self.inner.poll_ready(cx)) {
+                    Ok(()) => {
+                        // If the primary service becomes ready, drop the
+                        // fallback.
+                        drop(fallback.take().expect("fallback expected"));
+                        Poll::Ready(Ok(()))
+                    }
+                    Err(e) => {
+                        let (mut alt, target, predicate) =
+                            fallback.take().expect("fallback expected");
+                        // If the initial readiness failed in an expected way,
+                        // use the fallback service.
+                        if (predicate)(&e) {
```

Comment:
nit: the parentheses around a closure/function pointer are only necessary when calling it through a dotted field access (like `self.predicate`). it's not necessary here:
```suggestion
                        if predicate(&e) {
```